tests/components/version/test_sensor.py

```diff
@@ -1,31 +1,49 @@
 """The test for the version sensor platform."""
+from datetime import timedelta
 from unittest.mock import patch
 
 from pyhaversion import HaVersionSource, exceptions as pyhaversionexceptions
 import pytest
 
-from homeassistant.components.version.sensor import ALL_SOURCES
+from homeassistant.components.version.sensor import HA_VERSION_SOURCES
 from homeassistant.setup import async_setup_component
+from homeassistant.util import dt
+
+from tests.common import async_fire_time_changed
 
 MOCK_VERSION = "10.0"
 
 
 @pytest.mark.parametrize(
-    "source",
-    ALL_SOURCES,
+    "source,target_source,name",
+    (
+        (
+            ("local", HaVersionSource.LOCAL, "current_version"),
+            ("docker", HaVersionSource.CONTAINER, "latest_version"),
+            ("hassio", HaVersionSource.SUPERVISOR, "latest_version"),
+        )
+        + tuple(
+            (source, HaVersionSource(source), "latest_version")
+            for source in HA_VERSION_SOURCES
+            if source != HaVersionSource.LOCAL
+        )
+    ),
 )
-async def test_version_source(hass, source):
+async def test_version_source(hass, source, target_source, name):
     """Test the Version sensor with different sources."""
     config = {
         "sensor": {"platform": "version", "source": source, "image": "qemux86-64"}
     }
 
-    with patch("pyhaversion.version.HaVersion.version", MOCK_VERSION):
+    with patch("pyhaversion.version.HaVersion.get_version"), patch(
```

Comment:
Are we patching in the right place? Maybe patch the name in the `homeassistant.components.version.sensor.` imported path instead?

https://docs.python.org/3/library/unittest.mock.html#where-to-patch

Reply:
Both works (probably because `pyhaversion` is imported here) , but can change to the name imported in the sensor file instead :+1: